File: src/main/java/org/example/collection/validators/ValidatorMusicBandAlbumsCount.java
package org.example.collection.validators;

import org.example.exception.InvalidDataError;

/**
 * Валидатор количества альбомов музыкальной группы
 */
public class ValidatorMusicBandAlbumsCount implements Validator<Long> {
    /**
     * Проверяет число на соответствие диапазону
     *
     * @param along число
     * @throws InvalidDataError     если число не соответствует диапазону
     * @throws NullPointerException если число null
     */
    @Override
    public void valide(Long along) throws InvalidDataError {
        if (along == null) {
            throw new NullPointerException();
        }
        if (along <= 0) {
            throw new InvalidDataError();
        }
    }
}

File: src/main/kotlin/collection/validators/ValidatorMusicBandAlbumsCount.kt
package org.chousik.collection.validators

import org.chousik.exception.InvalidDataError


class ValidatorMusicBandAlbumsCount : IValidator<Long?> {

    override fun valide(t: Long?) {
        if (t == null) {
            throw NullPointerException()
        }
        if (t <= 0) {
            throw InvalidDataError()
        }
    }
}